File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& nums) {
          int minprices=nums[0];
        int profit=0;
        for(int i=1;i<nums.size();i++){
            if(nums[i]>minprices){
              profit = std::max(profit, nums[i] - minprices);
            }else{
                minprices=nums[i];
            }
        }
        return profit;
    }
};

// Synced seamlessly with LeetHub Pro
// Pro features: https://bit.ly/leethubpro | Free version: https://bit.ly/leethubv4
// Get it here: https://chromewebstore.google.com/detail/bcilpkkbokcopmabingnndookdogmbna

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] prices) {
        int maxprofit=0;
        int min= prices[0];
        for(int i=1;i<prices.length;i++){
            int cost=prices[i]-min;
            maxprofit= Math.max(maxprofit , cost);
            min= Math.min(min , prices[i]);
        }
        return maxprofit;
    }
}

// Synced seamlessly with LeetHub Pro
// Pro features: https://bit.ly/leethubpro | Free version: https://bit.ly/leethubv4
// Get it here: https://chromewebstore.google.com/detail/bcilpkkbokcopmabingnndookdogmbna